jormungandr-lib/src/crypto/key.rs

```diff
@@ -40,14 +40,34 @@ pub struct Identifier<A: AsymmetricPublicKey>(
 /// More info at the module documentation
 ///
 /// [`Identifier`]: ./struct.Identifier.html
-#[derive(Deserialize, Serialize)]
-pub struct SigningKey<A: AsymmetricKey>(
-    #[serde(
-        deserialize_with = "internal::deserialize_secret",
-        serialize_with = "internal::serialize_secret"
-    )]
-    pub(crate) SecretKey<A>,
-);
+
+pub struct SigningKey<A: AsymmetricKey>(pub(crate) SecretKey<A>);
+
+impl<A> Serialize for SigningKey<A>
+where
+    A: AsymmetricKey,
+    SecretKey<A>: Bech32,
+{
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        internal::serialize_secret(&self.0, serializer)
+    }
+}
+
+impl<'de, A> Deserialize<'de> for SigningKey<A>
+where
+    A: AsymmetricKey,
+    SecretKey<A>: Bech32,
+{
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        internal::deserialize_secret(deserializer).map(Self)
+    }
+}
```

Comment:
What is the improve of this change?

Reply:
Not all `SecretKey<A: AsymmetricKey>` can be serialized with `Bech32` now that this is restricted to fixed size data. I did not want to restrict the usage of `SigningKey` to just the keys that can be serialized / deserialized, so I added the bounds separately.

Reply:
Aha, I see, didn't notice the bounds though. Nice, thanks!